src/main/java/com/google/devtools/build/lib/syntax/Tuple.java

```diff
@@ -156,7 +156,7 @@ public int size() {
 
   @Override
   public Object[] toArray() {
-    return Arrays.copyOf(elems, elems.length);
+    return elems.length != 0 ? elems.clone() : elems;
```

Comment:
Without the copy, a caller might modify `elems`?

Reply:
* `elems.clone()` is identical to `Arrays.copyOf(elems, elems.length)` but makes optimizer life a litte easier
* empty array is not resizable, so sharing it is not an issue

Reply:
Indeed, thanks!
I'm slightly disappointed that `Arrays.copyOf` is not as fast.

Reply:
I think it is practically almost equally fast, I just changed this place mechanically, and can revert it back.